Sources/BraintreeShopperInsights/Models/CreateCustomerSessionMutationGraphQLBody.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+
+// swiftlint:disable nesting
+/// The POST body for the GraphQL mutation `CreateCustomerSession`
+struct CreateCustomerSessionMutationGraphQLBody: Encodable {
+    
+    var mutation: String
+    var variables: Variables
+    
+    init(request: BTCustomerSessionRequest) throws {
+        self.mutation = """
+            mutation CreateCustomerSession($input: CreateCustomerSessionInput!) {
+                createCustomerSession(input: $input) {
+                    sessionId
+                }
+            }
+            """
+        self.variables = Variables(request: request)
+    }
+    
+    struct Variables: Encodable {
+        
+        let input: InputParameters
+        
+        init(request: BTCustomerSessionRequest) {
+            self.input = InputParameters(
+                customer: Variables.InputParameters.Customer(
+                    hashedEmail: request.customer.hashedEmail,
+                    hashedPhoneNumber: request.customer.hashedPhoneNumber,
+                    paypalAppInstalled: request.customer.paypalAppInstalled,
+                    venmoAppInstalled: request.customer.venmoAppInstalled
+                ),
+                purchaseUnits: request.purchaseUnits?.map { purchaseUnit in
```

Comment:
Is graphQL okay if we call this map and pass `purchaseUnits` with an empty object? Not sure if we've had a chance to manually test that yet.

Reply:
ahh good catch, i haven't tested that case yet but i've updated to use `compactMap` to be on the safe side and added a unit test to cover it.